docs/guides/src/main/server/proxy.adoc

```diff
@@ -0,0 +1,28 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+
+<@tmpl.guide
+title="Proxy setup"
+summary="Understand how to configure Keycloak behind a reverse proxy"
+includedOptions="proxy proxy.*">
+
+All proxy configuration options have a `proxy` prefix. You can use different proxy modes depending the proxy and TLS termination in your specific environment.
+
+The `none` mode disables proxy support. It is the default mode.
+
+The `edge` mode enables communication through HTTP. This mode is suitable for deployments with a highly secure internal network where the reverse proxy keeps a secure connection (HTTP over TLS) with clients while communicating with Keycloak using HTTP.
+
+The `reencrypt` mode requires communication through HTTPS. This mode is suitable for deployments where internal communication between the reverse proxy and Keycloak should also be protected where different keys and certificates can be used on the reverse proxy as well as on Keycloak.
+
+The `passthrough` mode enables communication through HTTP or HTTPS. This mode is suitable for deployments where the reverse proxy is not terminating TLS, but only forwarding the requests to the Keycloak server so that secure connections between the server and clients are based on the keys and certificates used by the Keycloak server itself.
+
+Selecting the proxy mode is done at runtime. To select the proxy mode, run:
+
+<@kc.start parameters="--proxy <mode>"/>
```

Comment:
Not only the `start` command but it should also be possible to have options set when running `build`.

I'm not sure about the best way to communicate that, but perhaps we could have:

```
bin/kc.[bat|sh] [build|start] --proxy <mode>
```

Reply:
I used the start command instead of `build` here bc. it's a runtime option. I don't care mentioning also the build (or, fwiw, start-dev) command, let's just agree on a common format.

Reply:
added a new template for all commands in the latest version, see
<img width="740" alt="Screenshot 2021-12-17 at 17 12 19" src="https://user-images.githubusercontent.com/89905860/146574724-94ade04f-b594-4611-9b14-4b32790a010c.png">

Reply:
> I used the start command instead of `build` here bc. it's a runtime option. I don't care mentioning also the build (or, fwiw, start-dev) command, let's just agree on a common format.

Yeah, but I'm not sure too :) My main concern is not really to show all commands that support the option, but if we should have examples recommending `build` first or if the approach you are taking by using `start` with runtime and `build` with build props is just fine.

Your original proposal is shorter though.